config/settings/base.py

```diff
@@ -530,11 +530,6 @@
 METABASE_SITE_URL = "https://stats.inclusion.beta.gouv.fr"
 METABASE_SECRET_KEY = os.environ.get("METABASE_SECRET_KEY", "")
 
-# Once SIAE stats are released, all SIAE can view their data.
-# Until then, only whitelisted users can view them.
-RELEASE_STATS_SIAE = False
-STATS_SIAE_USER_PK_WHITELIST = json.loads(os.environ.get("STATS_SIAE_USER_PK_WHITELIST", "[]"))
```

Comment:
je vois que `PILOTAGE_DASHBOARDS_WHITELIST` est aussi comme cela, quelle était la logique pour prendre du JSON plutôt qu'une liste en CSV (par exemple) ?

Reply:
Euh... JSON est plus universel et se lit bien ?

Je ne comprends pas bien ta contre-proposition en fait. Comment proposerais-tu de stocker une liste de 3 valeurs dans un tel "CSV" ? Ce serait une ligne par valeur donc `1\n2\n3` ? J'ai du mal à voir comment ce serait plus lisible qu'un JSON `[1, 2, 3]`.

Côté conf clever c'est au final une simple string hein.

@rsebille

Reply:
Je n'avais pas spécialement de contre-proposition, c'est pas du nouveau code donc changer pour changer c'est #flemme.
Ma question était surtout pour savoir si il y avais une raison précise pour avoir pris du JSON, que ce soit à ces endroits, parce que c'est des listes, parce qu'on veux passer à du JSON pour la config, etc :).
C'est surement l'habitude qui parle mais je m'attendais à avoir ça sous la forme `1,2,3`, surtout pour une liste plate ^^.

Reply:
JSON permet universellement et facilement de convertir dans les deux sens un objet potentiellement complexe (ici une liste d'entiers) en une string plate (seule façon de stocker une variable dans la conf clever).

Si tu veux stocker `PILOTAGE_DASHBOARDS_WHITELIST = "1,2,3"` dans la conf clever alors tu devrais utiliser ici du code custom genre `[int(value) for value in PILOTAGE_DASHBOARDS_WHITELIST.split(",")]` ce qui est moins lisible qu'un simple `json.loads(PILOTAGE_DASHBOARDS_WHITELIST)`.

@rsebille

Reply:
j'ai pas d'avis sur la question, tout me va, mais je comprends d'où vient romain, j'ai souvent vu des paramètres séparés par des virgules dans des configs donc ça n'est pas moins courant que du JSON.

par contre là où je pense qu'on a une marge de progression dans un futur avec beaucoup de variables de config, serait de lire et valider le type de nos valeurs de conf avec du code custom un jour  (écrire un petit `conf.getintlist(PILOTAGE_DASHBOARDS_WHITELIST, [1, 2 ,3])` ne ferait pas forcément de mal pour dézinguer les valeurs de configuration erronées)